tools/bouncer_worker/src/lib/processMonitor.js

```diff
@@ -1,59 +1,139 @@
 const pidusage = require('pidusage');
 const processExists = require('process-exists');
+const si = require('systeminformation');
 const fs = require('fs');
 const logger = require('./logger');
-const { elastic } = require('./config').config;
+const Elastic = require('./elastic');
+const { enabled, memoryIntervalMS } = require('./config').config.processMonitoring;
+const elasticEnabled = require('./config').config.elastic.enabled;
 
-const processMonitor = {};
-processMonitor.logLabel = { label: 'PROCESSMON' };
+const ProcessMonitor = {};
+const logLabel = { label: 'PROCESSMONITOR' };
 
-const pidArray = [];
-let stats = [];
-let maxMemory = 0;
-let startMemory = 0;
+const informationDict = {};
+const workingDict = {};
+const pidSet = new Set();
+const permittedOS = ['linux', 'win32'];
 
-processMonitor.maxmem = async () => {
+const getCurrentOperatingSystem = async () => {
+	let data;
 	try {
-		const data = Number(fs.readFileSync('/sys/fs/cgroup/memory/memory.usage_in_bytes'));
-		if (pidArray.length > 0) {
-			stats = await pidusage(pidArray);
-			if (data > maxMemory) maxMemory = data;
+		data = await si.osInfo();
+	} catch (e) {
+		logger.error('Failed to get operating system information record', logLabel);
+	}
+	return data.platform;
+};
+const currentOSPromise = getCurrentOperatingSystem();
+
+const getDockerEnvironment = async () => {
+	const currentOS = await currentOSPromise;
+	if (currentOS === 'linux' && fs.existsSync('/.dockerenv')) {
+		return true;
+	}
+	return false;
+};
+const isDockerEnvironment = getDockerEnvironment();
+
+const getCurrentMemUsage = async () => {
+	let data;
+	let result;
+	try {
+		if (isDockerEnvironment) {
+			// required to get more accurate information in docker
+			result = Number(fs.readFileSync('/sys/fs/cgroup/memory/memory.usage_in_bytes'));
 		}
-		} catch (err) { logger.verbose(`[processMonitor][maxmem][error]: ${err}`, processMonitor.logLabel); }
+		data = await si.mem();
+		result = data.used;
+	} catch (e) {
+		logger.error('Failed to get memory information record', logLabel);
+		result = 0;
+	}
+	return result;
+};
+
+const maxmem = async () => {
+	let data;
+	let pids;
+
+	try {
+		data = await getCurrentMemUsage();
```

Comment:
as far as I can tell you don't overwrite data again, it's also not used outside of this scope. so it should just be a const here.
```suggestion
		const data = await getCurrentMemUsage();
```